communicator/vmturbocommunicator/server_message_handler.go

```diff
@@ -399,7 +443,15 @@ func (handler *MesosServerMessageHandler) NewMesosProbe(previousUseMap map[strin
 
 func (handler *MesosServerMessageHandler) monitorSlaveStatistics(s util.Slave, previousUseMap map[string]*util.CalculatedUse, mapTaskRes map[string]util.Statistics, mapSlaveUse map[string]*util.CalculatedUse, mapTaskUse map[string]*util.CalculatedUse, ports_slaves []string) error {
 	fullUrl := "http://" + util.GetSlaveIP(s) + ":" + handler.meta.SlavePort + "/monitor/statistics.json"
+
 	req, err := http.NewRequest("GET", fullUrl, nil)
+
+	req.Header.Add("content-type", "application/json")
+
+	if handler.meta.DCOS {
+		req.Header.Add("authorization", "token="+handler.meta.Token)
+	}
+
```

Comment:
What is the handler of token expiration??
Please create a method handleTokenExpiration , in which you call postRequest, and this method should be called in every get method, when you get exception in response